pkg/gofr/datasource/sql/sql_test.go

```diff
@@ -256,6 +257,50 @@ func Test_NewSQLMockWithConfig(t *testing.T) {
 	assert.NotNil(t, mockMetric)
 }
 
+var errSqliteConnection = errors.New("connection failed")
+
+func Test_sqliteLogs(t *testing.T) {
+	tests := []struct {
+		desc        string
+		status      string
+		err         error
+		expectedLog string
+	}{
+		{"sqlite connection in process", "connecting", nil, `connecting to 'test' database`},
+		{"sqlite connected successfully", "connected", nil, `connected to 'test' database`},
+		{"sqlite connection failure", "", errSqliteConnection,
+			`could not connect to database 'test', error: connection failed`},
+	}
+
+	for _, test := range tests {
+		if test.err == nil {
```

Comment:
Instead of having an `if-else` statement in test , it's better to break down it into two test methods one for error case and one for success case